### ISSUE TITLE
Kotest 4.1.1 doesn't discover tests (4.0.2 does that pretty well)

### ISSUE THREAD

**Issue by ispbox** [kotest/kotest#1544]
I tried to update my kotest 4.0.2 to 4.1.1, and got a problem that no unit tests are discovered (they are just grey in AS 4.0, and it's not possible to run them).

I've never had junit initialization as is mentioned in docs:
test {
    useJUnitPlatform()
}

I tried to add this but nothing changed.
Gradle version is 6.5.1. The only change is in version (from 4.0.2 to 4.1.1). Is it a temporary problem that will be fixed?

**Comment by ispbox** [comment]
I've searched through changelog and found new feature:

Feature: The Kotest IntelliJ plugin has gone final. The plugin requires 4.1. or higher of Kotest. https://plugins.jetbrains.com/plugin/14080-kotest

Does it mean that Kotest 4.1 and higher requires special plugin for running tests in AS 4.0? Previously everything worked without that plugin.

**Comment by ispbox** [comment]
4.0.6 is the last working version for me.
I've installed kotest plugin, but it looks like the problem with discovery is related to unknown symbols of DescribeSpec ("context" and "it"). What I'm doing wrong?

**Comment by sksamuel** [comment]
With 4.1.1 and the plugin you should be good.

**Comment by LeoColman** [comment]
>Does it mean that Kotest 4.1 and higher requires special plugin for running tests in AS 4.0?

Yes. If you don't want to use the plugin, you can execute tests via Gradle. I recommend the plugin tho

>I've installed kotest plugin, but it looks like the problem with discovery is related to unknown symbols of DescribeSpec ("context" and "it"). What I'm doing wrong?

This looks to me like Android Studio is doing it's magic and forgetting about how to compile the project. My usual approach to this is deleting `.idea`, deleting `.gradle` and then `invalidate and restart`

**Comment by ispbox** [comment]
Thanks a lot for a prompt answer!
The problem was that in one file with DescribeSpec I used mixture of "describe" and "context" regions. After changing to "describe" everywhere I got it fixed.

But there is still one minor issue. If I execute tests via Kotest plugin - everything is smooth. But if I try to run the same test from Android JUnit configuration (old one, that worked perfectly in Kotest 4.0.2), it fails with this strange message:

!!! JUnit version 3.8 or later expected:

java.lang.ClassNotFoundException: junit.framework.ComparisonFailure
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.intellij.rt.junit.JUnitStarter.junitVersionChecks(JUnitStarter.java:209)
	at com.intellij.rt.junit.JUnitStarter.canWorkWithJUnitVersion(JUnitStarter.java:195)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:49)

Is that an expected behavior? Should I forget about running Kotest without a plugin?

**Comment by sksamuel** [comment]
Yeah 4.1 requires the plugin because it's better all round

**Comment by sksamuel** [comment]
The old junit configurations won't work with 4.1.0
Also context should be gone/deprecated from DescribeSpec in favour of, as you say, nested describe scopes.

**Comment by ispbox** [comment]
> Yeah 4.1 requires the plugin because it's better all round

It depends on tasks. In my case I see that sometimes after fresh loading of a project, when I run a test, test framework quits unexpectedly. Probably it is because test is not built yet, but runner is going forward without waiting for a build. It happened several times, not every time, but old runner didn't have such an issue. Second run solves the problem.

I'll investigate this further, and probably will create a separate issue.

**Comment by elbaulp** [comment]
Hello, for me it does not work even after installing the plugin:

```
Jul 17, 2020 9:54:40 AM org.junit.platform.launcher.core.DefaultLauncher handleThrowable
WARNING: TestEngine with ID 'kotest' failed to discover tests
java.lang.NoClassDefFoundError: kotlin/time/TimeSource$Monotonic
	at io.kotest.core.engine.discovery.Discovery.scanUris(Discovery.kt:177)
	at io.kotest.core.engine.discovery.Discovery.access$scanUris(Discovery.kt:23)
	at io.kotest.core.engine.discovery.Discovery$fromClassPaths$2.invoke(Discovery.kt:32)
	at io.kotest.core.engine.discovery.Discovery$fromClassPaths$2.invoke(Discovery.kt:23)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at io.kotest.core.engine.discovery.Discovery.getFromClassPaths(Discovery.kt)
	at io.kotest.core.engine.discovery.Discovery.discover(Discovery.kt:55)
	at io.kotest.runner.junit.platform.KotestJunitPlatformTestEngine.discover(KotestJunitPlatformTestEngine.kt:81)
	at io.kotest.runner.junit.platform.KotestJunitPlatformTestEngine.discover(KotestJunitPlatformTestEngine.kt:24)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:177)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:164)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: java.lang.ClassNotFoundException: kotlin.time.TimeSource$Monotonic
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 16 more
```

**Comment by sksamuel** [comment]
Looks like you need 1.3.72 of kotlin.

**Comment by elbaulp** [comment]
Yep, already solve it. Thanks.
*-- Alejandro Alcalde - elbauldelprogramador.com
<http://elbauldelprogramador.com/en>*

**Comment by funct7** [comment]
I wish the docs were more explicit about this. Wasted hours searching, downloading the sample project, changing versions to find out what was wrong.

**Comment by sksamuel** [comment]
@funct7 what would you like the docs to be more explicit on exactly? Let me know and I'll make a PR to update them.

**Comment by funct7** [comment]
The [document on installation for Android](https://kotest.io/docs/quickstart#test-framework) doesn't mention anything about the Kotest IntelliJ plugin. Also the fact that `useJUnitPlatform()` seems to be deprecated also confuses users if they should go ahead and use it.

As this issue points out, none of the tests were being discovered, and I had to download the Android sample code and change the dependency version to verify that tests weren't being discovered above 4.1.x.

**Comment by sksamuel** [comment]
The plugin is only required for the play icon in intellij not for gradle.

I'll make that more clear.
